redisvl/index/storage.py

```diff
@@ -393,18 +394,30 @@ class HashStorage(BaseStorage):
     """Hash data type for the index"""
 
     def _validate(self, obj: Dict[str, Any]):
-        """Validate that the given object is a dictionary, suitable for storage
-        as a Redis hash.
+        """Validate that the given object is a dictionary suitable for storage
+        as a Redis hash, and the vector byte string is of correct datatype.
 
         Args:
             obj (Dict[str, Any]): The object to validate.
 
         Raises:
             TypeError: If the object is not a dictionary.
+            ValueError: if the vector byte string is not of correct datatype.
         """
         if not isinstance(obj, dict):
             raise TypeError("Object must be a dictionary.")
 
+        try:
+            byte_string = obj["prompt_vector"]
```

Comment:
specific to semantic caching?

Reply:
If we need for v1 we could remove this kind of validation logic for now

Reply:
That's leftover from an early attempt at adding dtypes. Removed it now.